android/src/main/kotlin/com/onegini/mobile/sdk/flutter/helpers/SdkError.kt

```diff
@@ -0,0 +1,119 @@
+package com.onegini.mobile.sdk.flutter.helpers
+
+import com.onegini.mobile.sdk.flutter.OneWelcomeWrapperErrors.*
+import com.onegini.mobile.sdk.flutter.OneWelcomeWrapperErrors
+import com.onegini.mobile.sdk.flutter.constants.Constants.Companion.RESPONSE_BODY
+import com.onegini.mobile.sdk.flutter.constants.Constants.Companion.RESPONSE_HEADERS
+import com.onegini.mobile.sdk.flutter.constants.Constants.Companion.RESPONSE_STATUS_CODE
+import com.onegini.mobile.sdk.flutter.constants.Constants.Companion.RESPONSE_URL
+import io.flutter.plugin.common.MethodChannel
+import okhttp3.Response
+
+class SdkError {
+    private val code: Int
+    private val message: String
+    private val details: MutableMap<String, Any> = mutableMapOf()
+
+    // Only error codes
+    constructor(
+        code: Int,
+        message: String?,
+    ) {
+        this.code = code
+        this.message = message ?: GENERIC_ERROR.message
+
+        setGenericDetails()
+    }
+
+    constructor(
+        wrapperError: OneWelcomeWrapperErrors
+    ) {
+        this.code = wrapperError.code
+        this.message = wrapperError.message
+
+        setGenericDetails()
+    }
+
+    // Error codes with userInfo
+    constructor(
+        code: Int,
+        message: String?,
+        info: Map<String, Any>?,
+    ) {
+        this.code = code
+        this.message = message ?: GENERIC_ERROR.message
+
+        setGenericDetails()
+        setInfoDetails(info)
+    }
+
+    constructor(
+        wrapperError: OneWelcomeWrapperErrors,
+        info: Map<String, Any>?,
+    ) {
+        this.code = wrapperError.code
+        this.message = wrapperError.message
+
+        setGenericDetails()
+        setInfoDetails(info)
+    }
+
+    // Error codes with httpResponse information
+    constructor(
+        code: Int,
+        message: String?,
+        httpResponse: Response?,
+    ) {
+        this.code = code
+        this.message = message ?: GENERIC_ERROR.message
+
+        setGenericDetails()
+        setResponseDetails(httpResponse)
+    }
+
+    constructor(
+        wrapperError: OneWelcomeWrapperErrors,
+        httpResponse: Response?,
```

Comment:
The only place this is called the response is not nullable. So let's change this constructor to not accept a nullable value.